crates/shim-sgx/src/main.rs

```diff
@@ -337,17 +345,24 @@ unsafe extern "C" fn main(
         1 => {
             // cssa == 0 already initialized the TCB
             let tcb = tcb.assume_init_mut();
+            let block = validate_block_ptr(block_ptr);
             handler::Handler::handle(
                 &mut ssas[0],
                 block.as_mut_slice(),
                 tcb,
                 _start as usize as _,
             )
         }
-        n => {
+        2 => {
+            let tcb = tcb.assume_init_mut();
+            let block = validate_block_ptr(block_ptr);
+            handler::Handler::finish(&mut ssas[1], Some(block.as_mut_slice()), tcb)
+        }
+        3 => {
             let tcb = tcb.assume_init_mut();
-            handler::Handler::finish(&mut ssas[n - 1], block.as_mut_slice(), tcb)
+            handler::Handler::finish(&mut ssas[2], None, tcb)
         }
+        _ => panic!("CSSA > 3"),
```

Comment:
Do we actually want to panic here? The panic handler at this point is an infinite loop, right? Is this the safest/most useful approach?

Reply:
It's safe, but not useful. I might change panic to loop on the `hlt` opcode in the future.